s3torchconnector/tst/e2e/dcp/test_e2e_s3_file_system.py

```diff
@@ -4,19 +4,315 @@
+    )
+
+    # Load using S3StorageReader
+    loaded_data_s3storagereader = {}
+    dcp_load(
+        loaded_data_s3storagereader,
+        S3StorageReader(
+            region,
+            s3_path_s3storagewriter,
+        ),
+    )
+
+    # Load using FsspecReader
+    loaded_data_fsspec = {}
+    dcp_load(
+        loaded_data_fsspec,
+        FsspecReader(
+            s3_path_fsspec,
+        ),
+    )
+
+    # Compare loaded data
+    assert set(loaded_data_s3storagereader.keys()) == set(
+        loaded_data_fsspec.keys()
+    ), "Loaded data keys do not match"
+
+    for key in loaded_data_s3storagereader.keys():
+        assert torch.allclose(
+            loaded_data_s3storagereader[key], loaded_data_fsspec[key]
+        ), f"Loaded tensors for key '{key}' do not match"
+        assert torch.allclose(
+            loaded_data_s3storagereader[key], test_data[key]
+        ), f"S3StorageReader: Loaded tensor for key '{key}' does not match original"
+        assert torch.allclose(
+            loaded_data_fsspec[key], test_data[key]
+        ), f"FsspecReader: Loaded tensor for key '{key}' does not match original"
+
+    print(
+        "Test passed: Both implementations saved and loaded data correctly, and the contents match."
+    )
+
+
+def test_dcp_when_multi_process_single_thread(checkpoint_directory):
+    multi_process_dcp_save_load(6, 1, checkpoint_directory)
+
+
+def test_dcp_when_multi_process_multiple_threads(checkpoint_directory):
+    multi_process_dcp_save_load(6, 4, checkpoint_directory)
+
+
+def test_dcp_save_non_existing_s3_uri(checkpoint_directory):
+    t1 = torch.randn(10)
+    region = checkpoint_directory.region
+    non_existing_s3_uri = "s3://non-existing-bucket/checkpoint"
+
+    with pytest.raises(CheckpointException) as s3_excinfo:
+        dcp_save(
+            {"random": t1},
+            S3StorageWriter(
+                region,
+                non_existing_s3_uri,
+                overwrite=True,
+            ),
+        )
+
+    with pytest.raises(CheckpointException) as fsspec_excinfo:
+        dcp_save(
+            {"random": t1},
+            FsspecWriter(
+                non_existing_s3_uri,
+                overwrite=True,
+                sync_files=False,
+            ),
+        )
+
+    # Assert that both exceptions are instances of CheckpointException
+    assert isinstance(
+        s3_excinfo.value, CheckpointException
+    ), "Using S3StorageWriter DCP should raise a CheckpointException"
+    assert isinstance(
+        fsspec_excinfo.value, CheckpointException
+    ), "Using FsspecWriter DCP should raise a CheckpointException"
+
+    print(
+        "Test passed: Both S3StorageWriter and FsspecWriter implementations raised the same CheckpointException."
+    )
+
+
+def test_dcp_load_non_existing_s3_uri(checkpoint_directory):
+    region = checkpoint_directory.region
+    non_existing_s3_uri = "s3://non-existing-bucket/checkpoint"
```

Comment:
Will that bucket name have meaning for S3 Express?

Reply:
No, will change this

Reply:
same as above comment